src/ucp/core/ucp_worker.c

```diff
@@ -1890,10 +1897,24 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
 
 static void ucp_worker_destroy_eps(ucp_worker_h worker)
 {
-    ucp_ep_ext_gen_t *ep_ext, *tmp;
+    ucp_ep_ext_gen_t *ep_ext, *ep_ext_tmp;
+    ucp_request_t *req, *req_tmp;
+
+    ucs_list_for_each_safe(req, req_tmp, &worker->close_reqs,
+                           send.disconnect.list) {
+        ucs_assert(req->send.ep->worker == worker);
+        ucp_ep_close_request_complete(req, UCS_ERR_CANCELED);
```

Comment:
maybe should be UCS_OK?

Reply:
I dont think so since user destroys worker before all related requests are completed.